driver/compile.go

```diff
@@ -14,45 +14,54 @@ import (
 	"go.uber.org/zap"
 )
 
+type Config struct {
+	TypeContext *resolver.Context
+	SortKey     string
```

Comment:
Nit: It’s unclear that this is a property of the reader, so I think you should document that or call it ReaderSortKey.